readme.markdown

```diff
@@ -431,19 +429,54 @@ which are also useful. Take a look at it for more information.
 
 ### Project structure
 
-There is nothing surprising about the recommended project structure. The advice
-resembles that of many other package managers.
+A Nimble project includes a *source directory*, containing a at most one
+primary source file, which shares the same name as the project itself (as well
+as the project's nimble file). In most cases this source directory will also be
+the root directory of the whole project. In all cases, the root directory will
+contain the .nimble file.
 
-| Directory     | Purpose                                |
-| ------------- | -------------------------------------- |
-| ``.``         | Root directory containing .nimble file.|
-| ``./src/``    | Project source code                    |
-| ``./tests/``  | Project test files                     |
-| ``./docs/``   | Project documentation                  |
+If the project includes additional source files, or if there is more than one
+primary (exported) module, they are all included in a single directory
+hierarchy within the source directory. In the case of libraries, this directory
+will have the same name as the project (see below for details).
 
-**Note:** Nimble will by default look for source files in ``.``, in order to
-use this layout you will need to specify ``srcDir = "src"`` in your .nimble
-file.
+The source directory can contain additional files and directories
+not involved in building the project, as long as they are excluded
+in the nimble file.
+
+Here's a sample one-module project directory:
+
+```
+.                       # The root directory of the project, also the source directory
+├── LICENSE
+├── README.md
+├── my_project.nim      # The primary source file
+├── my_project.nimble   # The project nimble file
+└── tests               # Another source directory, excluded in my_project.nimble
+    ├── nim.cfg
+    └── tests.nim
+```
+
+Here's a more complex example:
+
+```
+.                           # Root directory
+├── my_project.nimble       # Nimble file
+├── nim_src                 # Source directory
```

Comment:
I think this runs the risk of encouraging people to name their source dirs ``nim_src`` which we definitely do not want.